trace_replay/block_cache_tracer.cc

```diff
@@ -0,0 +1,188 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "trace_replay/block_cache_tracer.h"
+
+#include "db/db_impl/db_impl.h"
+#include "rocksdb/slice.h"
+#include "util/coding.h"
+#include "util/string_util.h"
+
+namespace rocksdb {
+
+namespace {
+const unsigned int kCharSize = 1;
+bool ShouldTraceReferencedKey(const TraceRecord& record) {
+  return (record.block_type == TraceType::kBlockTraceDataBlock) &&
+         (record.caller == BlockCacheLookupCaller::kUserGet ||
+          record.caller == BlockCacheLookupCaller::kUserMGet);
+}
+}  // namespace
+
+BlockCacheTraceWriter::BlockCacheTraceWriter(
+    Env* env, const TraceOptions& trace_options,
+    std::unique_ptr<TraceWriter>&& trace_writer)
+    : env_(env),
+      trace_options_(trace_options),
+      trace_writer_(std::move(trace_writer)) {
+  WriteHeader();
+}
+
+BlockCacheTraceWriter::~BlockCacheTraceWriter() { trace_writer_.reset(); }
```

Comment:
Seems you do not have to explicitly call `std::unique_ptr::reset` here.

Reply:
Removed.